library/src/blas2/rocblas_gemv.hpp

```diff
@@ -7,214 +7,40 @@
 #include "handle.h"
 #include "rocblas.h"
 
-template <typename T>
+template <typename T, typename U, typename V, typename W>
 rocblas_status rocblas_gemv_template(rocblas_handle    handle,
                                      rocblas_operation transA,
                                      rocblas_int       m,
                                      rocblas_int       n,
-                                     const T*          alpha,
-                                     const T*          A,
+                                     const U*          alpha,
+                                     const V*          A,
+                                     rocblas_int       offseta,
                                      rocblas_int       lda,
-                                     const T*          x,
+                                     rocblas_stride    strideA,
+                                     const V*          x,
+                                     rocblas_int       offsetx,
                                      rocblas_int       incx,
-                                     const T*          beta,
-                                     T*                y,
-                                     rocblas_int       incy)
+                                     rocblas_stride    stridex,
+                                     const U*          beta,
+                                     W*                y,
+                                     rocblas_int       offsety,
+                                     rocblas_int       incy,
+                                     rocblas_stride    stridey,
+                                     rocblas_int       batch_count)
 {
     //quick return
     if(!m || !n)
```

Comment:
Should this be `if(!m || !n || !batch_count)` ?

Reply:
good catch